mlflow/server/handlers.py

```diff
@@ -187,6 +188,38 @@ def _get_artifact_repo_mlflow_artifacts():
     return _artifact_repo
 
 
+def _get_trace_artifact_repo(trace_info: TraceInfo):
```

Comment:
Can we do the same approach as `get_artifact_handler`? It doesn't work?

Reply:
@harupy No, there is difference in how we handle repo and subpath;
* Run - reate an artifact repo at root URI, give subpath when calling `repo.download_artifacts(path)`.
```
mlflow-artifacts://{experiment_id}/{run_id}/artifacts
<--- repo's root artifact URI ----><--- subpath --->
``` 

* Trace - create an artifact repo with full trace specific URI (`traces/{request_id}`), pass nothing when calling `repo.download_trace_data()`.
```
mlflow-artifacts://{eperiment_id}/traces/{request_id}/artifacts
<----------------- repo's artifact URI ----------------------->
```

In other word, runs share the same artifact repo, while a trace create its own artifact repo. With this reason, we can't use `global` either.

Reply:
One question is - can we do the above approach for creating trace artifact repo? Maybe yes, but probably later🙂

Reply:
Thanks for the explanation! Did some manual debugging to see the difference.

> can we do the above approach for reusing trace artifact repo for traces?

SGTM!